app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -188,14 +189,9 @@ protected void onCreate(Bundle savedInstanceState) {
             if (Intent.ACTION_VIEW.equals(intent.getAction())) {
                 final String url = intent.getDataString();
 
-                if (Settings.getInstance(this).shouldShowFirstrun()) {
-                    pendingUrl = url;
-                    this.screenNavigator.addFirstRunScreen();
-                } else {
-                    boolean openInNewTab = intent.getBooleanExtra(IntentUtils.EXTRA_OPEN_NEW_TAB,
-                            false);
-                    this.screenNavigator.showBrowserScreen(url, openInNewTab, true);
-                }
+                boolean openInNewTab = intent.getBooleanExtra(IntentUtils.EXTRA_OPEN_NEW_TAB,
```

Comment:
this.screenNavigator.addFirstRunScreen(); is missed for this case?

Reply:
I think this is intended behavior change, details in #3454.
When the launch intent is VIEW, load and show the URL have higher priority, so here removed the addFirstRunScreen part and also pendingUrl cache.